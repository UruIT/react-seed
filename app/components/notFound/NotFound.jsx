import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
	<div>
		<h3>Page not found</h3>
		<Link to='/'>Return home</Link>
	</div>
);

export default NotFound;
