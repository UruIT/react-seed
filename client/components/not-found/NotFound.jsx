import React from 'react';
import { Link } from 'react-router-dom';
import links from '../../routes/links';
import styles from './not-found.scss';

const NotFound = () => (
	<div className={styles.container}>
		<h1>404</h1>
		<h2>Page Not Found</h2>
		<Link className={styles.link} to={links.index}>
			Go Home!
		</Link>
	</div>
);

export default NotFound;
