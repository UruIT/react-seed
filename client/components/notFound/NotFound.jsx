import React from 'react';
import { Link } from 'react-router-dom';
import styles from './not-found.scss';

const NotFound = () => (
	<div className={styles.container}>
		<h1>404</h1>
		<h2>Page Not Found</h2>
		<Link className={styles.link} to="/">
			Go Home!
		</Link>
	</div>
);

export default NotFound;
