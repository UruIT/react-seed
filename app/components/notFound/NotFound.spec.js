import React from 'react';
import NotFound from './NotFound';
import renderer from 'react-test-renderer';
import withContext from '../__mocks__/context-hoc';

test('<NotFound/>', () => {
	const ROUTER = {
		 router: {
			 history: {
				 createHref: jest.fn(),
				 push: jest.fn(),
				 replace: jest.fn()
			 }
		}
	};
	const NotFoundWithContext = withContext(ROUTER)(NotFound);

	const tree = renderer.create(
		<NotFoundWithContext />
	).toJSON();

	expect(tree).toMatchSnapshot();
});
