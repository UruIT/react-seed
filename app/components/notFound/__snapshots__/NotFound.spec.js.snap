// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound/> 1`] = `
<div>
  <h3>
    Page not found
  </h3>
  <a
    href={undefined}
    onClick={[Function]}
  >
    Return home
  </a>
</div>
`;
